pkg/build/api/validation/validation.go

```diff
@@ -239,7 +240,8 @@ func validateImageChange(imageChange *buildapi.ImageChangeTrigger) errs.Validati
 	if len(imageChange.Image) == 0 {
 		allErrs = append(allErrs, errs.NewFieldRequired("image"))
 	}
-	if len(imageChange.From.Name) == 0 {
+	var emptyObject kapi.ObjectReference
+	if imageChange.From == emptyObject {
```

Comment:
If from won't be set it'll be empty by default, so you need to check just the name, IMHO.